File: lollys-lotto-rust-sdk/src/instructions/admin/close_lollys_lotto.rs
use crate::instructions::*;

pub fn close_lollys_lotto(
    authority: &Pubkey,
    lollys_lotto: &Pubkey,
    event_emitter: &Pubkey,
) -> Instruction {
    let data = lollys_lotto::instruction::CloseLollysLotto.data();
    let accounts = lollys_lotto::accounts::CloseLollysLotto {
        authority: *authority,
        lollys_lotto: *lollys_lotto,
        event_emitter: *event_emitter,
        system_program: system_program::ID,
    }
    .to_account_metas(None);
    Instruction {
        program_id: lollys_lotto::ID,
        accounts,
        data,
    }
}

File: lollys-lotto-ts-sdk/src/codegen/instructions/closeLollysLotto.ts
import { TransactionInstruction, PublicKey, AccountMeta } from "@solana/web3.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import BN from "bn.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as borsh from "@coral-xyz/borsh" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as types from "../types" // eslint-disable-line @typescript-eslint/no-unused-vars
import { PROGRAM_ID } from "../programId"

export interface CloseLollysLottoAccounts {
  authority: PublicKey
  lollysLotto: PublicKey
  eventEmitter: PublicKey
  systemProgram: PublicKey
}

export function closeLollysLotto(
  accounts: CloseLollysLottoAccounts,
  programId: PublicKey = PROGRAM_ID
) {
  const keys: Array<AccountMeta> = [
    { pubkey: accounts.authority, isSigner: true, isWritable: true },
    { pubkey: accounts.lollysLotto, isSigner: false, isWritable: true },
    { pubkey: accounts.eventEmitter, isSigner: false, isWritable: true },
    { pubkey: accounts.systemProgram, isSigner: false, isWritable: false },
  ]
  const identifier = Buffer.from([135, 184, 63, 110, 146, 89, 76, 4])
  const data = identifier
  const ix = new TransactionInstruction({ keys, programId, data })
  return ix
}
